traits/has_traits.py

```diff
@@ -1111,12 +1111,13 @@ class HasTraits(CHasTraits, metaclass=MetaHasTraits):
 
     # -- Trait Definitions ----------------------------------------------------
 
-    #: An event fired when a new trait is dynamically added to the object
-    trait_added = Event(str)
+    #: An event fired when a new trait is dynamically added to the object. The
+    #: value is the name of the trait that was added.
+    trait_added = Event(Str())
 
     #: An event that can be fired to indicate that the state of the object has
-    #: been modified
-    trait_modified = Event
+    #: been modified.
+    trait_modified = Event()
```

Comment:
I guess we don't want to change behavior here by restricting this trait to `Event(Str())`?

Reply:
Unfortunately, I have no information on what the expected type of the event payload is (other than perhaps by analogy with `trait_added`. So no, I'm not going to introduce a type here.